### PR TITLE
Added setting domain on a service

### PR DESCRIPTION
**Pull request by jmaleonard**

### CODE REVIEW

File: lib/add-custom-resource.js [hunk 1]
```diff
@@ -113,8 +113,7 @@ module.exports = function addCustomResource(domainName, basePath ,deploymentId)
           '-', [
             this.serverless.service.service,
             this.options.stage,
-            this.options.region,
-            'custom-base-path-mapping-role'
```
Comment (dougmoscrop) [inline lib/add-custom-resource.js]: Thanks for this, was meaning to get to it

Comment (dougmoscrop) [reply lib/add-custom-resource.js]: Does it maybe need some kind of suffix? "cbpm" etc?

File: lib/remove-mapping.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 'use strict';
 
 module.exports = function removeMapping(basePath, domainName) {
+    if(!basePath){
```
Comment (dougmoscrop) [inline lib/remove-mapping.js]: Could this logic go in getBasePath?

Comment (jmaleonard) [reply lib/remove-mapping.js]: yeah, will move it there. got 2 more commits

File: lib/add-custom-resource.js (hunk 1 above)
Comment (jmaleonard) [reply lib/add-custom-resource.js]: Nope, I don't think so. there is a limit here (64 characters)

Comment (dougmoscrop) [reply lib/add-custom-resource.js]: lets delete the rolename entirely, since then CF will generate one.

Comment (jmaleonard) [reply lib/add-custom-resource.js]: Ok, lets test this

**Comment by jmaleonard** [comment]
My stack deployment worked.

**Comment by jmaleonard** [comment]
Are you going to accept this PR ? I am busy adding using my fork within my repo's. Would like to switch over to public npm

**Comment by dougmoscrop** [comment]
Yeah, I'm just fighting a cold.

You introduced a breaking change with the `domain.url` stuff, I think it's easy to write it to be compatible with both formats:

- domain === string ? basepath is `(none)` and that is the url.
- domain === object ? what you have

**Comment by jmaleonard** [comment]
ah yeah. Ok noted. Will submit a fix for that

**Comment by dougmoscrop** [comment]
I'll try to get to this soon, thanks.

**Comment by dougmoscrop** [comment]
I've incorporated (most of) these changes via https://github.com/dougmoscrop/serverless-plugin-custom-domain/pull/2 which will be released as 1.1.0

**Pull request closed without merging**